File: Leetcode/1056. Confusing Number/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn confusing_number(n: i32) -> bool {
        let str_n = n.to_string();
        let num_map: HashMap<char, char> = [('0', '0'), ('1', '1'), ('6', '9'), ('8', '8'), ('9', '6')].iter().cloned().collect();
        let rotated: Option<String> = str_n.chars().rev().map(|ch| num_map.get(&ch).copied()).collect();

        // If any digit invalid, rotated will be None, return false
        if rotated.is_none() {
            return false;
        }

        let rotated = rotated.unwrap();

        rotated!= str_n
    }
}

File: Leetcode/1056. Confusing Number/solution.py
class Solution:
    def confusingNumber(self, n: int) -> bool:
        stack = []
        str_n = str(n)
        num_map = {'0': '0', '1': '1', '6': '9', '8': '8', '9': '6'}

        for num in str_n[::-1]:
            if num not in num_map:
                return False
            else:
                stack.append(num_map[num])

        return "".join(stack) != str_n
